### PR TITLE
Close #70 - Soap 1.1 use SoapAction in RoundTripWithAction if possible

### LINKED ISSUE #70: wsdl's SoapAction not respected for Soap1.1
In Soap1.1, the soapAction header is set to the object's name even if the wsdl specifies the soapAction name.

For example, the following wsdl snippet
<operation name="ModifyObj">
<soap:operation soapAction="http://soap.example.com"/>
<input>
<soap:body use="literal"/>
</input>
<output>
<soap:body use="literal"/>
</output>
<fault name="WsValidationException">
<soap:fault name="WsValidationException" use="literal"/>
</fault>
</operation>

will generate 
	if err = p.cli.RoundTripWithAction("ModifyObj", α, &γ); err != nil {
		return nil, err
	}
	return &γ.Body.M, nil

instead, I should expect 
	if err = p.cli.RoundTripWithAction("http://soap.example.com", α, &γ); err != nil {
		return nil, err
	}
	return &γ.Body.M, nil

### PR DESCRIPTION
**Pull request by adriantam**
Otherwise, use the default object name

### CODE REVIEW

**Comment by fiorix** [comment]
Do you mind refactoring this code a bit to have a single parameters object for the template and a switch to decide which soap action (1.1 or 1.2) to use?

File: wsdlgo/testdata/w3example1.golden [hunk 1]
```diff
@@ -51,7 +51,7 @@ func (p *getEndorsingBoarderPortType) GetEndorsingBoarder(α *GetEndorsingBoarde
 			M GetEndorsingBoarderResponse `xml:"GetEndorsingBoarderResponse"`
 		}
 	}{}
-	if err = p.cli.RoundTripWithAction("GetEndorsingBoarder", α, &γ); err != nil {
```
Comment (Fank) [inline wsdlgo/testdata/w3example1.golden]: I think this should be `example.com`

File: wsdlgo/testdata/w3example1.golden [hunk 2]
```diff
@@ -51,7 +51,7 @@ func (p *getEndorsingBoarderPortType) GetEndorsingBoarder(α *GetEndorsingBoarde
...
+	if err = p.cli.RoundTripWithAction("http://www.snowboard-info.com/EndorsementSearch", α, &γ); err != nil {
```
Comment (Fank) [inline wsdlgo/testdata/w3example1.golden]: I think this should be `example.com`

Comment (fiorix) [reply wsdlgo/testdata/w3example1.golden]: Fair point, doesn't make sense to have this domain here but I already merged. I'm happy to merge another PR to fix this. Sorry about that

**Pull request merged**